sw/device/mask_rom/mask_rom_start.S

```diff
@@ -0,0 +1,270 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey_memory.h"
+
+/**
+ * Mask ROM Interrupt Vector
+ */
+  .section .vectors, "ax"
+  .option push
+
+  // Disable RISC-V instruction compression: we need all instructions to
+  // be exactly word wide in the interrupt vector.
+  .option norvc
+
+  // Disable RISC-V linker relaxation, as it can compress instructions at
+  // link-time, which we also really don't want.
+  .option norelax
+
+/**
+ * `_mask_rom_interrupt_vector` is the boot-defined interrupt vector for Ibex,
+ * for the Mask ROM.
+ *
+ * The Mask ROM does not use interrupts, so there are only entries for Ibex's
+ * non-maskable interrupts, and for the hardware exception handler.
+ *
+ * Interrupt vectors in Ibex have 32 entries for 32 possible interrupts. The
+ * vector must be 256-byte aligned, as Ibex's vectoring mechanism requires that.
+ * Ibex uses the instruction directly after the boot interrupt vector when
+ * starting execution from reset, which we choose to make look like an extra
+ * entry.
+ *
+ * Thus this vector has exactly 33 4-byte entries.
+ *
+ * More information about Ibex's interrupts can be found here:
+ *   https://ibex-core.readthedocs.io/en/latest/exception_interrupts.html
+ */
+  .balign 256
+  .globl _mask_rom_interrupt_vector
+  .type _mask_rom_interrupt_vector, @function
+_mask_rom_interrupt_vector:
+
+  // exception Handler and user software interrupt
+  .extern mask_rom_exception_handler
+  j mask_rom_exception_handler
+  // supervisor software interrupt
+  unimp
+  // reserved
+  unimp
+  // machine software interrupt handler
+  unimp
+
+  // user timer interrupt
+  unimp
+  // supervisor timer interrupt
+  unimp
+  // reserved
+  unimp
+  // machine timer interrupt handler
+  unimp
+
+  // user external interrupt
+  unimp
+  // supervisor external interrupt
+  unimp
+  // reserved
+  unimp
+  // machine external interrupt handler
+  unimp
+
+  // reserved
+  unimp
+  unimp
+  unimp
+  unimp
+
+  // vendor interrupts: on Ibex interrupt ids 30-16 are the "fast" interrupts
+  .rept 15
+  unimp
+  .endr
+
+  // vendor interrupts: on Ibex interrupt id 31 is for non-maskable interrupts
+  .extern mask_rom_nmi_handler
+  j mask_rom_nmi_handler
+
+  // Reset handler (mtvec + 0x80)
+  j _mask_rom_start_boot
+
+  // Set size so vector can be disassembled
+  .size _mask_rom_interrupt_vector, .-_mask_rom_interrupt_vector
+
+  .option pop
```

Comment:
Nit: Consider adding a comment that says that this pops relaxation and rv32c compression.